engine/schema/src/main/java/com/cloud/dc/dao/ClusterDaoImpl.java

```diff
@@ -168,16 +168,26 @@ public List<HypervisorType> getAvailableHypervisorInZone(Long zoneId) {
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Returns distinct (HypervisorType, CPUArch) pairs from clusters in the given zone,
+     * excluding clusters with {@link HypervisorType#External}.
+     *
+     * @param zoneId the zone ID to filter by, or {@code null} to include all zones
+     * @return list of unique hypervisor type and CPU architecture pairs
+     */
     @Override
     public List<Pair<HypervisorType, CPU.CPUArch>> listDistinctHypervisorsArchAcrossClusters(Long zoneId) {
```

Comment:
I feel the method name should reflect the change to avoid confusion and wrong usage

Reply:
> @shwstppr can you check the new changes with name change